### PR TITLE
2199: Adding read only permission in /conf folder for

### PR DESCRIPTION
**Pull request by rajeevyadav2**
Issue:
acs-commons-email-service does not have read permissions in /conf directory. Due to this email templates in  /conf are not getting loaded and thus email service fails.

Fix details:
acs-commons-email-service user has been prvided read permission in /conf folder so that it can load the email templates.

### CODE REVIEW

**Comment by rajeevyadav2** [comment]
> Thanks for opening this pull request! The maintainers of this repository would appreciate it if you would update the CHANGELOG.md file with this pull request.

done

**Comment by HitmanInWis** [comment]
Anyone have any idea why the CI checks are not all running?

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 - #2146 - POI exception generating Excel file with too many references
 - #2178 - Worked around a POI exception with MCP Asset Folder Creator, due to the underlying bundle upgrading from POI v3.x->POI v4.x in 6.5.3 (addresses #2177 & #2162)
 - #2185 - fix empty iconpicker and fontawesome files
-
+- #2199 - Read permisions on /conf folder for acs-commons-email-service user
```
Comment (HitmanInWis) [inline CHANGELOG.md]: Please add an empty line below this

File: ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0" 
+jcr:primaryType="sling:OsgiConfig"
```
Comment (HitmanInWis) [inline ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml]: Please indent this file similar to the other OSGi configs

File: ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (HitmanInWis) [inline ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml]: Let's rename the file to EnsureServiceUser-EmailService.xml to be consistent with other files.

File: ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0" 
+jcr:primaryType="sling:OsgiConfig" 
+principalName="acs-commons/acs-commons-email-service" 
+operation="add" 
+ensure-immediately="{Boolean}true" 
+aces="[type=allow;privileges=jcr:read;path=/conf/global]" />
```
Comment (HitmanInWis) [inline ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml]: I tested this on a fresh instance of AEM 6.4.4 and on deployment I do not see the permissions added to /conf/global.  Can't say why - the config looks correct and I even see it on the configMgr felix interface.

The other existing EnsureServiceUser configs seem to be adding their permissions just fine, the service itself is working fine.

I tried restarting AEM to see if the permissions would be picked up on restart, and I still see no rep:policy on /conf/global.

Could the issue be that there is no rep:policy below /conf/global in the first place (on a clean server)?

**Comment by HitmanInWis** [comment]
```
27.02.2020 13:43:56.973 *ERROR* [CM Event Dispatcher (Fire ConfigurationEvent: pid=com.adobe.acs.commons.users.impl.EnsureServiceUser.29e95a0a-66d0-4008-9b0d-901238be0971)] com.adobe.acs.commons.users.impl.EnsureServiceUser Unable to ensure Service User [ acs-commons/acs-commons-email-service ]
com.adobe.acs.commons.users.impl.EnsureAuthorizableException: [ acs-commons-email-service ] is an System User provided by AEM or ACS AEM Commons. You cannot ensure this user.
```

**Comment by HitmanInWis** [comment]
Looks like this is b/c `com.adobe.acs.commons.users.impl.ProtectedAuthorizables#PRINCIPAL_NAMES` contains the `acs-commons-email-service` user.  I dont have enough context as to why that is important (to know if removing it will be a problem or not).

Note that both `acs-commons-remote-assets-service` and `acs-commons-file-fetch-service` are ACS Commons service users, but neither is included in `com.adobe.acs.commons.users.impl.ProtectedAuthorizables#PRINCIPAL_NAMES`

**Comment by rajeevyadav2** [comment]
> ProtectedAuthorizables
Thanks @HitmanInWis for details. Is there any other way to provide permissions to users listed in PRINCIPAL_NAMES? We can remove email-service user from PRINCIPAL_NAMES so that we can assign permissions using ensure service. What's your thought on this?

File: CHANGELOG.md (hunk 1 above)
Comment (rajeevyadav2) [reply CHANGELOG.md]: @HitmanInWis  Added

File: ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml (hunk 4 above)
Comment (rajeevyadav2) [reply ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml]: I removed the user from PRINCIPAL_NAMES and it should be good now. :)

File: ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml (hunk 3 above)
Comment (rajeevyadav2) [reply ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml]: Done. 👍

File: ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml (hunk 2 above)
Comment (rajeevyadav2) [reply ui.apps/src/main/content/jcr_root/apps/acs-commons/config/com.adobe.acs.commons.users.impl.EnsureServiceUser-email-service.xml]: Done. 👍

**Comment by HitmanInWis** [comment]
Looks like the `com.adobe.acs.commons.users.impl.ProtectedAuthorizables#PRINCIPAL_NAMES` mechanism was created by https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/906 in an effort to prevent ACLs being unexpectedly changed for an existing user.  I think we'd need someone like @davidjgonzalez or @andrewmkhoury to validate that this change isnt going to open undue risk or perhaps cause certain permissions set otherwise (either `rep:policy` nodes in acs-aem-commons, or set manually on existing AEM servers) to be lost.

**Comment by HitmanInWis** [comment]
@rajeevyadav2 thinking more on this... I wonder if EnsureServiceUser is even going to be a thing going forward, since Im not sure if the new Cloud Service will support this.  Perhaps we should just add a `rep:policy` node to the repo instead, and thus avoid this EnsureServiceUser issue you were hitting all together.

**Comment by rajeevyadav2** [comment]
> @rajeevyadav2 thinking more on this... I wonder if EnsureServiceUser is even going to be a thing going forward, since Im not sure if the new Cloud Service will support this. Perhaps we should just add a `rep:policy` node to the repo instead, and thus avoid this EnsureServiceUser issue you were hitting all together.

@HitmanInWis 
Adding rep:policy leaves consumers to add it in their repository. Either we should document this step in Email Service Feature or else we should inject this rep:policy as part of acs commons codebase. Should I go ahead and add rep:policy in /conf in this repo?

Also, Want to understand if this feature is not supported in AEM as cloud service?

**Comment by HitmanInWis** [comment]
Sorry for no being clear - yes Im advocating for adding the `rep:policy` node to the repo, which is how a lot of other permissions are being done.  I think the only concern would be to ensure it's tested in 6.3/6.4/6.5 so we don't mess anything up from OOTB permissions and folders.

**Pull request closed without merging**